src/EA.Weee.Web/Areas/Producer/Mappings/ToViewModel/EditContactDetailsMap.cs

```diff
@@ -0,0 +1,73 @@
+﻿namespace EA.Weee.Web.Areas.Producer.Mappings.ToViewModel
+{
+    using EA.Prsd.Core.Mapper;
+    using EA.Weee.Core.DirectRegistrant;
+    using EA.Weee.Core.Organisations;
+    using EA.Weee.Core.Organisations.Base;
+    using EA.Weee.Core.Shared;
+    using EA.Weee.Web.Areas.Producer.ViewModels;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    public class EditContactDetailsMap : IMap<SmallProducerSubmissionData, EditContactDetailsViewModel>
+    {
+        private readonly IMapper mapper;
+
+        public EditContactDetailsMap(IMapper mapper)
+        {
+            this.mapper = mapper;
+        }
+
+        public EditContactDetailsViewModel Map(SmallProducerSubmissionData source)
+        {
+            var externalOrganisationType = MapOrganisationType(source.OrganisationData.OrganisationType);
+            var businessAddressData = MapBusinessAddress(source);
+
+            var additionalContactDetails = source.CurrentSubmission.AdditionalCompanyDetailsData.Select(c => new AdditionalContactModel()
```

Comment:
@stevelee-civica  
 here you need to populate the EditContactDetailsViewModel ContactDetailsViewModel based on the SmallProducerSubmissionData.SmallProducerSubmissionHistoryData.ContactAddressData and ContactData properties.